test/client.test.js

```diff
@@ -2370,11 +2370,80 @@ describe('ACC Client', function () {
             // the in-memory cache of the client.
+            let callLengthOld = storage.getItem.mock.calls.length;
+            client = await Mock.makeClient({ storage: storage });
+            client._transport.mockReturnValueOnce(Mock.LOGON_RESPONSE);
+            await client.NLWS.xtkSession.logon();
+            client._transport.mockReturnValueOnce(Mock.GET_NMS_EXTACCOUNT_SCHEMA_RESPONSE);
+            await client.getSchema("nms:extAccount");
+            // Here we can't simply check the length of mock calls since there're for "lastCleared"
+            // We check inside the new calls, since the creation of the second client, there's one 
+            // for the schema "nms:extAccount"
+            let callLengthNew = storage.getItem.mock.calls.length;
+            expect(storage.getItem.mock.calls[callLengthNew-1][0]).toMatch("cache.XtkEntityCache$xtk:schema|nms:extAccount");
+        })
+
+        it("Should clear storage if necessary", async () => {
+            const sdkVersion = sdk.getSDKVersion().version;
+            const mockSdkVersionItemKey    = "acc.js.sdk.0.0.0.acc-sdk:8080.cache.Hello";
+            const otherItemKey             = "other.Hello";
+            const map = {};
+            const storage = {
+                "acc.js.sdk.0.0.0.acc-sdk:8080.cache.Hello": "0.0.0.World",
+                "other.Hello": "other.World",
+                getItem: jest.fn((key) => map[key]),
+                setItem: jest.fn((key, value) => map[key] = value),
+                removeItem: jest.fn((key) => { delete map[key] })
+            }
+
+            let client = await Mock.makeClient({ storage: storage });
+
+            // Remove item not matches to the current sdk version and begins with "acc.js.sdk"
+            expect(storage.removeItem.mock.calls.length).toBe(1);
+            expect(storage.removeItem.mock.calls[0][0]).toMatch(mockSdkVersionItemKey);
+
+            client._transport.mockReturnValueOnce(Mock.LOGON_RESPONSE);
+            await client.NLWS.xtkSession.logon();
+            client._transport.mockReturnValueOnce(Mock.GET_NMS_EXTACCOUNT_SCHEMA_RESPONSE);
+            await client.getSchema("nms:extAccount");
+            const currentSdkVersionItemKey = "acc.js.sdk."+sdkVersion+".acc-sdk:8080.cache.XtkEntityCache$xtk:schema|nms:extAccount";
+            expect(storage.setItem.mock.calls[0][0]).toMatch(currentSdkVersionItemKey);
+
+            // Remove ONLY the item not matches to the current sdk version but not the one matches
+            // Now simulate reusing the local storage to make sure that we remove ONLY the item 
+            // not matches to the current sdk version.
+            // We need a new client to make sure we do not reuse the in-memory cache of the client.
+            // We use the same test method as the one we use in the previous test "Should cache XML in storage"
+            let callLengthOld = storage.getItem.mock.calls.length;
```

Comment:
This variable does not seem to be used. Can you remove it?

Reply:
done